src/decode.rs

```diff
@@ -170,7 +170,9 @@ fn decode_node<I: Iterator<Item = u8>, J: Jet>(
                     None => return Err(Error::EndOfStream),
                     Some(true) => Ok(CommitNode::jet(context, J::decode(bits)?)),
                     Some(false) => {
-                        return Err(Error::ParseError("we don't yet support const words"))
+                        let depth = decode_natural(bits, Some(32))?;
```

Comment:
AFAIU, `decode_natural` [fails if the input is longer than 32 bit](https://github.com/BlockstreamResearch/rust-simplicity/blob/master/src/decode.rs#L330). Therefore `bound = Some(32)` and `bound = None` are equivalent. We should use the latter.

Reply:
Sure.

Reply:
Lol, you had me going. We fail if the *number* is longer than 32 bits. But here we actually need to bound the number at 32. The fuzzer found that if I remove this `Some(32)` then it's possible to try to read a depth of 157 (or any crazy value) which triggers a panic down the line.

Reply:
Oh. I will update the documentation of `decode_natural`. It is confusing what the bound is.

Reply:
The offending program was `"5d5d325e3332"` ... let's try to remember to add a unit test with this!